x/genaccounts/doc.go

```diff
@@ -0,0 +1,9 @@
+/*
```

Comment:
What is the motivation for having this as a module? Is it solely to be able to call `ExportGenesis`? I guess I'm not directly opposed to this, but it always made sense in my mind to have any genesis logic close to the application definition (ie. `gaia`).

If we can keep this general and modular enough, then I suppose it's fine. But it's not really atm because of vesting logic.

Reply:
> If we can keep this general and modular enough, then I suppose it's fine. But it's not really atm because of vesting logic.

Yeah I'd like to abstract this module further to achieve this level of generalization, having it in a separate module is just the first step. Even as is though it's really nice to have all the account functionality packaged up so that if you do want to change the genesis account type, its very clear what needs to be forked and modified as it's all packaged in one place. 

But basically the everything fits into the module convention quite nicely for set of genesis accounts. Here we use the module pattern to call `ExportGenesis` - but also InitGenesis and ValidateGenesis. Previously all this stuff was just tangled as a part of `gaia/init/...` and `gaia/app/genesis.go`

Reply:
> Even as is though it's really nice to have all the account functionality packaged up so that if you do want to change the genesis account type, its very clear what needs to be forked and modified as it's all packaged in one place.

++

Reply:
See comment below - I agree with these goals, but they would seem to me to be better fulfilled by making this a submodule of `x/auth` instead.

Reply:
That was my initial thinking too @cwgoes.